ARGUS/ARGUS_UNet/utils/ARGUS_visualization.py

```diff
@@ -0,0 +1,206 @@
+                plt.imshow(tmpV, cmap="gray")
+                subplot_num += 1
+            plt.subplot(2, num_subplots, num_subplots)
+            plt.title(f"label")
+            tmpV = test_data["label"][0, 0, :, :]
+            for c in range(self.num_classes):
+                tmpV[0, c] = c
+            plt.imshow(tmpV)
+            subplot_num += 1
+
+            # Indent by one plot
+            subplot_num = num_subplots + 1
+            prob_shape = test_outputs[0].shape
+            prob_total = np.zeros(prob_shape)
+            for run_num in range(num_runs):
+                prob = np.empty(prob_shape)
+                run_output = test_outputs[run_num]
+                for c in range(self.num_classes):
+                    itkProb = itk.GetImageFromArray(run_output[c, :, :])
+                    imMathProb = tube.ImageMath.New(itkProb)
+                    imMathProb.Blur(5)
+                    itkProb = imMathProb.GetOutput()
+                    prob[c] = itk.GetArrayFromImage(itkProb)
+                pmin = prob.min()
+                pmax = prob.max()
+                prange = pmax - pmin
+                prob = (prob - pmin) / prange
+                prob[class_artery] = prob[class_artery] * artery_prior
+                class_array = np.argmax(prob, axis=0)
+                done = False
+                while not done:
+                    done = True
+                    count = np.count_nonzero(class_array > 0)
+                    while count < artery_min_size:
+                        prob[class_artery] = prob[class_artery] * 1.05
+                        class_array = np.argmax(prob, axis=0)
+                        count = np.count_nonzero(class_array == class_artery)
+                        done = False
+                    while count > artery_max_size:
+                        prob[class_artery] = prob[class_artery] * 0.95
+                        class_array = np.argmax(prob, axis=0)
+                        count = np.count_nonzero(class_array == class_artery)
+                        done = False
+                denom = np.sum(prob, axis=0)
+                denom = np.where(denom == 0, 1, denom)
+                prob = [prob[c] / denom for c in range(self.num_classes)]
+                prob_total += prob
+                for c in range(self.num_classes):
+                    plt.subplot(2, num_subplots, subplot_num)
+                    plt.title(f"Class " + str(c))
+                    tmpV = prob[c, :, :]
+                    plt.imshow(tmpV, cmap="gray")
+                    subplot_num += 1
+
+            prob_total = prob_total / num_runs
+            subplot_num = num_subplots * 2 - 1
+            plt.subplot(2, num_subplots, subplot_num)
+            plt.title(f"Ensemble")
+            tmpV = prob_total[class_artery, :, :]
+            plt.imshow(tmpV, cmap="gray")
+            subplot_num += 1
+
+            class_array = np.argmax(prob_total, axis=0)
```

Comment:
The clean-up and labeling (here to line 206) will also have to go into a loop, for each class > 0.